properties/basic.py

```diff
@@ -861,9 +870,13 @@ def validate(self, instance, value):
         try:
             floatval = float(value)
             if not self.cast and abs(value - floatval) > TOL:
-                self.error(instance, value)
+                self.error(
+                    instance=instance,
+                    value=value,
+                    extra='Not within tolerance range of {}.'.format(TOL),
+                )
         except (TypeError, ValueError):
-            self.error(instance, value)
+            self.error(instance, value, extra='Cannot cast to float.')
```

Comment:
I never really realized this existed... I guess it mainly applies to complex floats?